File: CPE 185/LAB/Lab 3/NicksProject/gpiotest.c
#include <stdio.h>
#include <wiringPi.h>

//pin number declarations
const int ledPin = 17;
const int buttonPin = 27;

int main(void)
{
    //setup stuff:
    wiringPiSetupGpio(); //initialize wiring pi
    
    pinMode(ledPin, OUTPUT);
    pinMode(buttonPin, INPUT);
    
    printf("C GPIO program running! Press control C to quit.\n");
    
    while(1)
    {
        if(digitalRead(buttonPin))
            digitalWrite(ledPin, HIGH);
        else
            digitalWrite(ledPin, LOW);
    }
    
    return 0;

}

File: CPE 185/LAB/Lab 3/SorivProject/gpiotest.py
# External module imports
import RPi.GPIO as GPIO
import time

# Pin Definitons:
ledPin = 17 # Broadcom pin 17 (PI pin 11)
buttonPin = 27 # Broadcom pin 27 (PI pin 13)

# Pin Setup:
GPIO.setmode(GPIO.BCM)
GPIO.setup(ledPin, GPIO.OUT)
GPIO.setup(buttonPin, GPIO.IN)

# Initial state for LED:
GPIO.output(ledPin, GPIO.LOW)
print("Simple Python GPIO Start! Press CTRL+C to exit")
try:
    while 1:
        if GPIO.input(buttonPin):
            GPIO.output(ledPin, GPIO.HIGH)
            print("1")
            time.sleep(0.2)
        else:
            GPIO.output(ledPin, GPIO.LOW)
            
except KeyboardInterrupt: # If CTRL+C is pressed, exit and clean
    GPIO.cleanup() # cleanup all GPIO